rasa/core/utils.py

```diff
@@ -471,13 +471,13 @@ def replace_floats_with_decimals(obj: Union[List, Dict], round_digits: int = 9)
     """
     if isinstance(obj, list):
         for i in range(len(obj)):
-            obj[i] = replace_floats_with_decimals(obj[i])
+            obj[i] = replace_floats_with_decimals(obj[i], round_digits)
         return obj
     elif isinstance(obj, dict):
         for j in obj:
-            obj[j] = replace_floats_with_decimals(obj[j])
+            obj[j] = replace_floats_with_decimals(obj[j], round_digits)
         return obj
-    elif isinstance(obj, float):
+    elif isinstance(obj, float) or isinstance(obj, Decimal):
```

Comment:
the `isinstance(obj, Decimal)` part is in case the values were already converted?
Could you please add this case to the unit tests?

Reply:
Yep, in some cases, the value is already cast to Decimal. I added a test case as part of the commit here: https://github.com/RasaHQ/rasa/pull/5131/files#diff-5e57b5d5bd2748445fd012f8dd6b5305R102